werkzeug/useragents.py

```diff
@@ -28,8 +28,8 @@ class UserAgentParser(object):
         ('netbsd', 'netbsd'),
         ('openbsd', 'openbsd'),
         ('freebsd', 'freebsd'),
-        (r'x11|lin(\b|ux)?', 'linux'),
         ('(sun|i86)os', 'solaris'),
+        (r'x11|lin(\b|ux)?', 'linux'),
```

Comment:
Because ```r'x11|lin\b|ux)?'``` might match the ```x11``` in a user agent string for SunOS like this:
```Mozilla/5.0 (X11; SunOS i86pc; rv:38.0) Gecko/20100101 Firefox/38.0``` and result in parsing the platform as ```linux```. Moving it after solaris results in recognizing ```solaris``` as the correct platform.